cli/cmd/plugin/ui/web/tanzu-ui/src/shared/components/wizard/UseUpdateTabStatus.hooks.tsx

```diff
@@ -0,0 +1,18 @@
+import { useEffect, useRef } from 'react';
+import { FieldErrors } from 'react-hook-form';
+
+type UpdateTabStatus = (currentStep: number | undefined, validForm: boolean) => void;
+
+function UseUpdateTabStatus(errors: FieldErrors, currentStep: number | undefined, updateTabStatus: UpdateTabStatus) {
+    const hasError = Object.keys(errors).length > 0;
+    const prevHasErrorRef = useRef<boolean>(false);
+
+    useEffect(() => {
+        if (updateTabStatus && prevHasErrorRef.current !== hasError) {
+            updateTabStatus(currentStep, !hasError);
+        }
+        prevHasErrorRef.current = hasError;
```

Comment:
nit: since I assume updateTabStatus is always non-null, it might make more sense to put this assignment inside the above block (because outside the block the two values are the same, so there's no need for an assignment). I think the code might read a little more easily that way, but it would obviously work either way.

Reply:
make sense. moved the line in the if block.